pkg/algorithm/cell_allocation.go

```diff
@@ -78,6 +79,70 @@ func buddyAlloc(
 	return false
 }
 
+// after buddyAlloc failed to allocate cells,
+// try to split a higher level cell to get current level cells.
+func splitAlloc(
+	cell *cellBindingPathVertex,
+	freeList ChainCellList,
+	freeCellNum map[CellLevel]int32,
+	currentLevel CellLevel,
+	suggestedNodes common.Set,
+	ignoreSuggestedNodes bool,
+	bindings map[api.CellAddress]*PhysicalCell) bool {
+
+	var splittableCell Cell
+	splittableNum := map[CellLevel]int32{}
+	for i := CellLevel(len(freeList)); i >= CellLevel(1); i-- {
+		// calculate splitable number
+		splittableNum[i] = int32(len(freeList[i])) - freeCellNum[i]
+		if i < CellLevel(len(freeList)) && splittableCell != nil {
+			splittableNum[i] += splittableNum[i+1] * int32(len(splittableCell.GetChildren()))
+		}
+		// iterate higher level cell
+		if splittableCell == nil && len(freeList[i]) > 0 {
+			splittableCell = freeList[i][0]
+		} else if splittableCell != nil {
+			splittableCell = splittableCell.GetChildren()[0]
+		}
+		// check safety
+		if splittableNum[i] < 0 {
+			panic(fmt.Sprintf("VC Safety Broken: level %v cell is unsplittable", i))
```

Comment:
maybe we should print cell address in the panic log as well, as we cannot see cell chain in this function

Reply:
added